src/celutil/utf8.cpp

```diff
@@ -961,8 +961,8 @@ static size_t greekChunkLength(const std::string& str)
         npos = true;
     }
 
-    if (sp != 0 && isdigit(str[sp - 1]))
-        while(sp != 0 && isdigit(str[sp - 1])) sp--;
+    if (sp != 0 && str[sp - 1] >= 0 && isdigit(str[sp - 1]))
```

Comment:
you are doing something strange here.

Reply:
I ran into a crash here with MSVC (debug). passing a negative value to isdigit aborts the application.

Reply:
then the correct way it to cast to uchar:
```
if (sp != 0 && isdigit(static_cast<unsigned char>(str[sp - 1])))
    while(sp != 0 && isdigit(static_cast<unsigned char>(str[sp - 1]))) sp--;
```